[v2.8] K8s November 2023 patches

vardhaman22 opened the pull request:
**skipping the october patches, since november patches are released**
related issues: https://github.com/rancher/rancher/issues/43074, https://github.com/rancher/rancher/issues/43130 ,https://github.com/rancher/rancher/issues/43204, https://github.com/rancher/rancher/issues/43637, https://github.com/rancher/rancher/issues/43639, https://github.com/rancher/rancher/issues/43199

- adds k8s v1.25.16, v1.26.11, v1.27.8 versions for rke1
- for all the new versions uses etcd v3.5.9 and rke-tools:v0.1.96
- adds nginx v1.9.4 template and use it in all newly introduced versions.
- for nginx v1.9.4 update ingress-nginx/kube-webhook-certgen  image to v20231011-8b53cabe0 [ref](https://github.com/kubernetes/ingress-nginx/blob/5583f90c7f3525b6267747d6b07541c01218a9ea/deploy/static/provider/cloud/deploy.yaml#L591)
- also includes the latest changes of dev-v2.8 branch (needed because a few more changes went in for september patches)
- bumped calico to v3.26.3 for new k8s versions
- bumped k8s dns images to 1.22.28 for new k8s versions
- included hyperkube-base fix https://github.com/rancher/rancher/issues/43199

brooksn committed:
v1.25.15+rke2r1 v1.26.10+rke2r1 v1.27.7+rke2r1

Signed-off-by: Brooks Newberry <brooks@newberry.com>

brooksn committed:
remove rke2 sep patches

Signed-off-by: Brooks Newberry <brooks@newberry.com>

vardhaman22 committed:
added nginx v1.9.4 template

superseb committed:
Bump rke-tools to v0.1.96

superseb committed:
go generate

superseb committed:
Merge pull request #1258 from superseb/v28_update_rketools_v0196

[dev-v2.8] Update rke-tools to v0.1.96

Commit:
[dev-v2.8] K3s patch release October (#1251)

* update v1.27.7-rc1+k3s1, v1.26.10-rc1+k3s1 and v1.25.15-rc1+k3s1

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* remove v1.27.6+k3s1, v1.26.9+k3s1 and v1.25.14+k3s1

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.27.7-rc2+k3s1, v1.26.10-rc2+k3s1 and v1.25.15-rc2+k3s1

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.27.7+k3s1, v1.26.10+k3s1 and v1.25.15+k3s1

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

---------

Signed-off-by: Johnatas <johnatasr@hotmail.com>

brooksn committed:
go generate

Signed-off-by: Brooks Newberry <brooks@newberry.com>

brooksn committed:
Merge pull request #1254 from brooksn/rke2-patch-2023-oct-2.8

[dev-v2.8] RKE2 patch releases

brooksn committed:
v1.25.15+rke2r2 v1.26.10+rke2r2 v1.27.7+rke2r2

Signed-off-by: Brooks Newberry <brooks@newberry.com>

brooksn committed:
go generate

Signed-off-by: GitHub <noreply@github.com>

johnatasr committed:
[dev-v2.8] K3s2 patch release  (#1263)

* update to v1.27.7-rc1+k3s2, v1.26.10-rc1+k3s2 and v1.25.15-rc1+k3s2

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.27.7-rc2+k3s2, v1.26.10-rc2+k3s2 and v1.25.15-rc2+k3s2

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.27.7-rc3+k3s2, v1.26.10-rc3+k3s2 and v1.25.15-rc3+k3s2

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* update to v1.27.7+k3s2, v1.26.10+k3s2 and v1.25.15+k3s2

Signed-off-by: Johnatas <johnatasr@hotmail.com>

* go generate

Signed-off-by: Johnatas <johnatasr@hotmail.com>

---------

Signed-off-by: Johnatas <johnatasr@hotmail.com>

brooksn committed:
Merge pull request #1261 from brooksn/rke2-oct-2023-r2-v2.8

[v2.8] RKE2 October patch release 2

snasovich reviewed (changes requested):
@vardhaman22 , please update `rke-tools` to `v0.1.96`. While at it, could you please also update it in previous versions as done per https://github.com/rancher/kontainer-driver-metadata/pull/1258 (in fact, please bring in other changes that got into `dev-v2.8` branch after we branched for `dev-v2.8-2023-10-patches` - e.g. RKE2/k3s R2 releases).

vardhaman22 commented:
> @vardhaman22 , please update `rke-tools` to `v0.1.96`. While at it, could you please also update it in previous versions as done per #1258 (in fact, please bring in other changes that got into `dev-v2.8` branch after we branched for `dev-v2.8-2023-10-patches` - e.g. RKE2/k3s R2 releases).

rebased my branch with dev-v2.8 and also updated rke-tools image to v0.1.96 in the latest changes.

vardhaman22 committed:
k8s november 2023 patches

vardhaman22 committed:
bump calico to v3.26.3 in k8s nov patches

vardhaman22 committed:
bump hyperkube image for nov 2023 patches

vardhaman22 committed:
updated rke default k8s version

kinarashah committed:
Merge pull request #1267 from rancher/dev-v2.8

[DNM pending release notes] [v2.8] KDM release for v2.8.0

kinarashah commented on pkg/rke/templates/templates.go:
```diff
@@ -117,8 +118,10 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			">=1.27.0-rancher1-1":                      calicov3261,
-			">=1.26.0-rancher1-1 <1.27.0-rancher1":     calicov3250,
-			">=1.25.0-rancher1-1 <1.26.0-rancher1-1":   calicov3241,
+			">=1.26.11-rancher1 <1.27.0-rancher1":      calicov3261,
```
`">=1.27.0-rancher1-1"` and `">=1.26.11-rancher1 <1.27.0-rancher1"` both are for  `calicov3261`, could you update this to `">=1.26.11-rancher1"`?

vardhaman22 replied:
updated

kinarashah commented on pkg/rke/templates/templates.go:
```diff
@@ -150,8 +153,10 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			">=1.27.0-rancher1-1":                      canalv3261,
-			">=1.26.0-rancher1-1 <1.27.0-rancher1":     canalv3250,
-			">=1.25.0-rancher1-1 <1.26.0-rancher1-1":   canalv3241,
+			">=1.26.11-rancher1 <1.27.0-rancher1":      canalv3261,
```
same as calico

vardhaman22 replied:
updated

vardhaman22 committed:
bump k8s dns node images to 1.22.28 for nov 2023 patches

vardhaman22 committed:
go generate

kinarashah reviewed (approved):
- confirmed the system images for existing k8s versions are now the same as released data 
- confirmed new versions for rke-tools, etcd, kube-dns, calico/canal, nginx 
- confirmed default k8s version bump to v1.27.8 
- confirmed version conditions for addons in templates.go 
- branch is rebased on latest `release-v2.8`

snasovich commented on pkg/rke/templates/templates.go:
```diff
@@ -116,9 +117,10 @@ var TemplateIntroducedRanges = map[string][]string{
-			">=1.27.0-rancher1-1":                      calicov3261,
-			">=1.26.0-rancher1-1 <1.27.0-rancher1":     calicov3250,
-			">=1.25.0-rancher1-1 <1.26.0-rancher1-1":   calicov3241,
+			">=1.26.11-rancher1":                       calicov3261,
```
Are we positive we're not going to hit any issues with upgrade given how templates here reference v3.26.1 vs. v3.26.3 in 2.7 PR (https://github.com/rancher/kontainer-driver-metadata/pull/1257/files#diff-98a74f75bbfd8f2ab7594591f9e71d268da7423281099e275c61bf283848928fR119)? I understand the templates may be the same but I'm concerned about names mismatch.

The same applies to Canal.

@kinarashah

kinarashah replied:
Ya I think RKE does the diff based on templates so it'd be mostly fine, but @snasovich good catch, we should update it to avoid confusion and make it consistent overall. Since `v3.26.1` already exists in release-v2.8, @vardhaman22 could we update everywhere to use `calicov3261`?

vardhaman22 replied:
updated the 2.7 PR with the changes.... no changes needed in this PR

snasovich replied:
@kinarashah , @vardhaman22 , thank you!

snasovich reviewed (commented):
One comment/concern to review. Otherwise LGTM. Similar to 2.7 PR, updated the description to only reference 2.8 issues.
